### ISSUE TITLE
go Insecure instead of Bogus on unknown algorithm

### ISSUE THREAD

**Issue by Habbie** [PowerDNS/pdns#4254]
Right now http://dnsviz.net/d/community.hallard.me/dnssec/ validates as Bogus for me, due to their use of DSA (algo 3) which we do not (and will not) support.

From https://tools.ietf.org/html/rfc4035#section-5.2

> If the validator does not support any of the algorithms listed in an
>    authenticated DS RRset, then the resolver has no supported
>    authentication path leading from the parent to the child.  The
>    resolver should treat this case as it would the case of an
>    authenticated NSEC RRset proving that no DS RRset exists, as
>    described above.

**Comment by pieterlexis** [comment]
Fixed in #5463

**Comment by pieterlexis** [comment]
```
Jun 27 09:46:55 [1]  hallard.me: Wants DNSSEC processing, NO auth data in query for DS
Jun 27 09:46:55 [1]  hallard.me: Found cache hit for DS: 51188 3 2 73f25123cdd714d12c5b36a392fd6b4b6c499ff0cf93a86225d1aff03231f1b7[ttl=86399] 
Jun 27 09:46:55 [1]  hallard.me: updating validation state with cache content for hallard.me to Secure
Jun 27 09:46:55 Discarding DS 51188 because we don't support algorithm number 3
Jun 27 09:46:55 [1] : New state for hallard.me is Insecure
```
